src/store/index.js

```diff
@@ -34,25 +38,76 @@ export default new Vuex.Store({
       state.collections = await state.connection.db.listCollections().toArray()
     },
     async find({ state }, query = {}) {
-      state.totalNumberOfResults = null
+      if (state.collectionRecords[state.collectionName]) {
+        const query_serialization = JSON.stringify(query)
+
+        if (state.collectionRecords[state.collectionName].query === query_serialization) {
+          state.activeRecords = state.collectionRecords[state.collectionName]
+          state.totalNumberOfResults = state.activeRecords.result.length
+          return
+        }
+      }
+
+      state.totalNumberOfResult = null
       const collection = state.connection.db.collection(state.collectionName)
       const cursor = await collection.find(query)
+
       cursor.count((err, result) => {
         if (err) return
-
         state.totalNumberOfResults = result
       })
-      state.records = await cursor.toArray()
+
+      const result = await cursor.toArray()
+
+      state.activeRecords = {
```

Comment:
why do we also keep active records? do we need it when we store `state.collectionRecords[state.collectionName]`?